libs/providers/rocketflag/src/lib/rocketflag-provider.ts

```diff
@@ -0,0 +1,141 @@
+import type { EvaluationContext, Provider, JsonValue, ResolutionDetails, Logger } from '@openfeature/web-sdk';
+import { ErrorCode, ProviderEvents, StandardResolutionReasons } from '@openfeature/web-sdk';
+import { EventEmitter } from 'events';
+
+export interface UserContext {
+  cohort?: string;
+}
+
+export interface FlagStatus {
+  enabled: boolean;
+}
+
+interface RocketFlagClient {
+  getFlag(flagKey: string, userContext: UserContext): Promise<FlagStatus>;
+}
+
+/**
+ * A helper function to fetch a flag, update the cache, and notify OpenFeature of changes.
+ * It's defined within the factory's scope to access the client, cache, emitter, and logger.
+ */
+const fetchFlagAndUpdateCache = (
+  // Parameters required for the fetch operation
+  flagKey: string,
+  defaultValue: boolean,
+  context: EvaluationContext,
+  cacheKey: string,
+  // State managed by the factory's closure
+  client: RocketFlagClient,
+  cache: Map<string, ResolutionDetails<boolean>>,
+  emitter: EventEmitter,
+  logger?: Logger,
+) => {
+  const userContext: UserContext = {};
+  const { targetingKey } = context;
+
+  if (targetingKey && typeof targetingKey === 'string' && targetingKey !== '') {
+    userContext.cohort = targetingKey;
+  }
+
+  client
+    .getFlag(flagKey, userContext)
+    .then((flagStatus) => {
+      const details: ResolutionDetails<boolean> = {
+        value: flagStatus.enabled,
+        reason: userContext.cohort ? StandardResolutionReasons.TARGETING_MATCH : StandardResolutionReasons.DEFAULT,
```

Comment:
`TARGETING_MATCH` should indicate why a value was evaluated the way it was. In this case, it doesn't look like you'll have enough information in the response to tell.

Reply:
I'm not sure I understand, but I will try to answer.

If the user provides a cohort value in the context, then the resolution reason will be a result of targeting match. As to _why_ the API responded that way, that information is not made available in the API response.